wandb/cli/cli.py

```diff
@@ -857,6 +851,283 @@ def _parse_settings(settings):
+    "using 'python' to run .py files and the default shell (specified by "
+    "environment variable $SHELL) to run .sh files. If passed in, will override the entrypoint value passed in using a config file.",
+)
+@click.option(
+    "--version",
+    "-v",
+    metavar="VERSION",
+    help="Version of the project to run, as a Git commit reference for Git projects.",
+)
+@click.option(
+    "--param-list",
+    "-P",
+    metavar="NAME=VALUE",
+    multiple=True,
+    help="A parameter for the run, of the form -P name=value. Provided parameters that "
+    "are not in the list of parameters for an entry point will be passed to the "
+    "corresponding entry point as command-line arguments in the form `--name value`",
+)
+@click.option(  # todo: maybe take these out it's confusing with the docker image stuff
+    "--docker-args",
+    "-A",
+    metavar="NAME=VALUE",
+    multiple=True,
+    help="A `docker run` argument or flag, of the form -A name=value (e.g. -A gpus=all) "
+    "or -A name (e.g. -A t). The argument will then be passed as "
+    "`docker run --name value` or `docker run --name` respectively. ",
+)
+@click.option(
+    "--experiment-name",
+    envvar="WANDB_NAME",
+    help="Name of the experiment under which to launch the run. If not "
+    "specified, 'experiment-id' option will be used to launch run. If passed in, will override the name passed in using a config file.",
+)
+@click.option(
+    "--entity",
+    "-e",
+    metavar="<str>",
+    default=None,
+    help="Name of the target entity which the new run will be sent to. Defaults to using the entity set by local wandb/settings folder."
+    "If passed in, will override the entity value passed in using a config file.",
+)
+@click.option(
+    "--project",
+    "-p",
+    metavar="<str>",
+    default=None,
+    help="Name of the target project which the new run will be sent to. Defaults to using the project set by local wandb/settings folder."
+    " If passed in, will override the project value passed in using a config file.",
+)
+@click.option(
+    "--resource",
+    "-r",
+    metavar="BACKEND",
+    default="local",
+    help="Execution resource to use for run. Supported values: 'local'."
+    " If passed in, will override the resource value passed in using a config file."
+    " Defaults to 'local'.",
+)
+@click.option(
+    "--docker-image",
+    "-d",
+    default=None,
+    metavar="DOCKER IMAGE",
+    help="Specific docker image you'd like to use. In the form name:tag."
+    " If passed in, will override the docker image value passed in using a config file.",
+)
+@click.option(
+    "--config",
+    "-c",
+    metavar="FILE",
+    help="Path to JSON file (must end in '.json') or JSON string which will be passed "
+    "as config to the compute resource. The exact content which should be "
+    "provided is different for each execution backend. See documentation for layout of this file.",
+)
+def launch(
```

Comment:
We should add `@display_error` and fix the cli logger so we don't dump big stacktraces to the terminal when things fail.